src/sources/soundsourcemodplug.cpp

```diff
@@ -0,0 +1,211 @@
+#include "sources/soundsourcemodplug.h"
+
+#include "metadata/trackmetadata.h"
+#include "util/timer.h"
+
+#include <QFile>
+
+#include <stdlib.h>
+#include <unistd.h>
+
+/* read files in 512k chunks */
+#define CHUNKSIZE (1 << 18)
+
+namespace Mixxx {
+
+namespace {
+
+QString getModPlugTypeFromUrl(QUrl url) {
+    const QString type(SoundSource::getTypeFromUrl(url));
+    if (type == "mod") {
+        return "Protracker";
+    } else if (type == "med") {
+        return "OctaMed";
+    } else if (type == "okt") {
+        return "Oktalyzer";
+    } else if (type == "s3m") {
+        return "Scream Tracker 3";
+    } else if (type == "stm") {
+        return "Scream Tracker";
+    } else if (type == "xm") {
+        return "FastTracker2";
+    } else if (type == "it") {
+        return "Impulse Tracker";
+    } else {
+        return "Module";
+    }
+}
+
+// identification of modplug module type
+enum ModuleTypes {
+    NONE = 0x00,
+    MOD = 0x01,
+    S3M = 0x02,
+    XM = 0x04,
+    MED = 0x08,
+    IT = 0x20,
+    STM = 0x100,
+    OKT = 0x8000
+};
+
+} // anonymous namespace
+
+const SINT SoundSourceModPlug::kChannelCount = 2; // always stereo
+const SINT SoundSourceModPlug::kFrameRate = 44100; // always 44.1 kHz
+
+QList<QString> SoundSourceModPlug::supportedFileExtensions() {
+    QList<QString> list;
+    // ModPlug supports more formats but file name
+    // extensions are not always present with modules.
+    list.push_back("mod");
+    list.push_back("med");
+    list.push_back("okt");
+    list.push_back("s3m");
+    list.push_back("stm");
+    list.push_back("xm");
+    list.push_back("it");
+    return list;
+}
+
+unsigned int SoundSourceModPlug::s_bufferSizeLimit = 0;
+
+// reserve some static space for settings...
```

Comment:
That comment is kind of wrong now ;)
The modplug settings are not saved in a static class member anymore, which is correct. Comment should be moved to line 70 (static buffer size limit) or removed completely.